src/modules/todo-funtionalities.js

```diff
@@ -0,0 +1,79 @@
+import Storage from './storage.js';
+import Todo from './todo.js';
+
+const toDoList = document.querySelector('.toDoList');
+
+// My To Do list
+let toDoListsArray = [];
+
+const displayTodos = () => {
+  // Get the todos from local storage
+  toDoListsArray = Storage.getDataFromLocalStorage();
+  // Sort the table of todos
+  const sortedTodoLists = toDoListsArray.sort((a, b) => a.index - b.index);
+  let myToDos = '';
+
+  sortedTodoLists.forEach((todo) => {
+    const checked = todo.completed ? 'checked' : '';
+    const complete = todo.completed ? ' completed' : '';
+    myToDos += `
+    <li class="list-item" id="${todo.index}">
+          <article class="status-title">
+            <input data-id="${todo.index}" class="status" type="checkbox" ${checked} title="check!">
+            <input data-id="${todo.index}" type="text" class="toDoEditInput ${complete}" value="${todo.description}" title="Click on '${todo.description}' to EDIT it.">
+          </article>
+          <article class="actions">
+          <span class="material-symbols-outlined remove">delete</span>
+          <span class="material-symbols-outlined verticalDots">more_vert</span>
+          </article>
+          
+          
+        </li>`;
+  });
+  toDoList.innerHTML = myToDos;
+};
```

Comment:
Every time the user clicks on a todo [(line 55),](https://github.com/kensteph/js-best-practices/pull/1/files#diff-bfe9874d239014961b1ae4e89875a6155667db834a410aaaa2ebe3cf89820556R55) clear all completed [(line 72),](https://github.com/kensteph/js-best-practices/pull/1/files#diff-bfe9874d239014961b1ae4e89875a6155667db834a410aaaa2ebe3cf89820556R72) add [(line 47),](https://github.com/kensteph/js-best-practices/pull/1/files#diff-66f382bc8d3e7090cc4593021be73a4224ca7425f4d0d7b4e0416a70fa569688R47) remove [(line 61)](https://github.com/kensteph/js-best-practices/pull/1/files#diff-66f382bc8d3e7090cc4593021be73a4224ca7425f4d0d7b4e0416a70fa569688R61) or edit [(line 74)](https://github.com/kensteph/js-best-practices/pull/1/files#diff-66f382bc8d3e7090cc4593021be73a4224ca7425f4d0d7b4e0416a70fa569688R74) one of them you're executing the display todos function, which is fine. However, in this function, you are rendering **all todos** every time is called inside the todo list container this might result in performance issues on low-range phones. 

I kindly suggest considering using [document fragments](https://developer.mozilla.org/en-US/docs/Web/API/DocumentFragment) to speed up the rendering process and also to avoid calling this function on certain events, for instance, when the user removes an element instead of rendering all the todos again you can simply use the [.remove()](https://www.javascripttutorial.net/dom/manipulating/remove-a-dom-element/) method to get rid of the element from the UI, and after that remove it from the array using the [.filter()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) method, also when the user adds a todo you can simply just [append](https://developer.mozilla.org/en-US/docs/Web/API/Node/appendChild) it to the container.